### PR TITLE
Update CLI11 to version 2.4.2

### LINKED ISSUE #1457: CLI11 warnings on BlueOS+CUDA.
These warnings were reported by the Ares team and they would like them to be corrected. It looks like they are building on BlueOS with CUDA enabled.

/usr/gapps/bdiv/blueos_3_ppc64le_ib_p9/clang-ibm-16.0.6-cuda-11.8.0-gcc-11.2.1/axom/0.10.0-cuda/include/axom/CLI11.hpp(1001): warning #20236-D: pragma "diag_suppress" is deprecated, use "nv_diag_suppress" instead

/usr/gapps/bdiv/blueos_3_ppc64le_ib_p9/clang-ibm-16.0.6-cuda-11.8.0-gcc-11.2.1/axom/0.10.0-cuda/include/axom/CLI11.hpp(1005): warning #20236-D: pragma "diag_default" is deprecated, use "nv_diag_default" instead

### PR DESCRIPTION
**Pull request by BradWhitlock**
This PR updates CLI11 to version 2.4.2 in order to get some changes that should avoid the nvcc warnings reported in #1457. 

Resolves #1457.

### CODE REVIEW

File: src/docs/sphinx/dev_guide/updating_tpls.rst [hunk 1]
```diff
@@ -245,6 +245,22 @@ instructions on how to update a built-in TPL are:
...
+line processing. Axom packages the library in a header-only format. The CLI11.hpp
+header file must be downloaded from a released version of CLI11 since the source
+code repository provides sources in separate header files. The patch file contains
+many small changes that can be summarized as follows:
+
+#. Add "namspace axom {" near the top of the file.
+#. Add "} // namspace axom" near the bottom of the file.
+#. Move "#pragma once" from below the copyright to the top of the file.
+#. Replace "CLI::" with "axom::CLI::".
+#. Replace "Success" with "CLI11_Success". This avoids a symbol collision with X11.
```
Comment (kennyweiss) [inline src/docs/sphinx/dev_guide/updating_tpls.rst:262]: 👍

**Review by kennyweiss — APPROVED** [review]
Thanks @BradWhitlock !

File: src/thirdparty/axom/CLI11.patch [hunk 2]
```diff
@@ -1,128 +1,222 @@
-diff --git a/src/thirdparty/axom/CLI11.hpp b/src/thirdparty/axom/CLI11.hpp
-index 68244d386..1dc6eb9ae 100644
---- a/src/thirdparty/axom/CLI11.hpp
-+++ b/src/thirdparty/axom/CLI11.hpp
-@@ -159,6 +159,8 @@
- 
- // From StringTools.hpp:
+--- CLI11.hpp	2024-10-28 16:39:59.708552000 -0700
++++ CLI11_new.hpp	2024-10-28 17:10:57.111210000 -0700
```
Comment (white238) [inline src/thirdparty/axom/CLI11.patch:2]:
```suggestion
+++ CLI11.hpp	2024-10-28 17:10:57.111210000 -0700
```

**Review by white238 — APPROVED** [review]

File: src/thirdparty/axom/CLI11.patch (hunk 2 above)
Comment (BradWhitlock) [reply src/thirdparty/axom/CLI11.patch]: Done

**Pull request merged**